JoelFaldin/ServerSort-MuniCrud-Backend
# Feat/add auth route handlers

# Add auth route handlers and services

## Description 🗒️✏️

Added route handlers related to auth services. Login, logout, and the process to reset user password.

## How to test it? 🎯⚗️🔬

- Install the dependencies using `pnpm i`.
- For prisma, run `pnpx prisma migrate dev --name init`, then run `pnpx prisma generate`. Aditionally, you can start a local page to inspect the tables using `pnpx prisma studio`, but is not completely necessary.
- Create an user using Postman or Insomnia, then try to log in/log out of the app, should work correctly.
- With the user created, try to recover your password. Send the rut and email to `http://localhost:3000/api/auth/password`, and it should send an email to the users email address. The message will contain a link, click it and you should be able to enter a page dedicated to update the password. Enter and repeat the new password, and it should update the password correctly.

<div align="center">

![image](https://github.com/user-attachments/assets/8074f36a-0be7-4846-be8a-494fee251178)
![image](https://github.com/user-attachments/assets/587484c8-0191-4364-bf09-fe3e2c7a2c9f)


</div>
